pkg/engine/handlers/validation/validate_pss.go

```diff
@@ -119,15 +131,100 @@ func (h validatePssHandler) Process(
-		return resource, handlers.WithResponses(
-			engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks),
+		ruleResponse := engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks)
+		allowExisitingViolations := rule.HasValidateAllowExistingViolations()
+		if engineutils.IsUpdateRequest(policyContext) && allowExisitingViolations {
+			logger.V(4).Info("is update request")
+			priorResp, err := h.validateOldObject(ctx, logger, policyContext, resource, rule, engineLoader, exceptions)
+			if err != nil {
+				return resource, engineapi.RuleError(rule.Name, engineapi.Validation, "failed to validate old object", err, rule.ReportProperties)
+			}
+
+			if ruleResponse.Status() == priorResp.Status() {
+				logger.V(3).Info("skipping modified resource as validation results have not changed", "oldResp", priorResp, "newResp", ruleResponse)
+				if ruleResponse.Status() == engineapi.RuleStatusPass {
+					return resource, ruleResponse
+				}
+				return resource, engineapi.RuleSkip(rule.Name, engineapi.Validation, "skipping modified resource as validation results have not changed", rule.ReportProperties)
+			}
+			logger.V(4).Info("old object response is different", "oldResp", priorResp, "newResp", ruleResponse)
+		}
+
+		return resource, ruleResponse
+	}
+}
+
+func (h validatePssHandler) validateOldObject(
+	ctx context.Context,
+	logger logr.Logger,
+	policyContext engineapi.PolicyContext,
+	resource unstructured.Unstructured,
+	rule kyvernov1.Rule,
+	engineLoader engineapi.EngineContextLoader,
+	exceptions []*kyvernov2.PolicyException,
+) (*engineapi.RuleResponse, error) {
+	if policyContext.Operation() != kyvernov1.Update {
+		return nil, nil
+	}
+
+	newResource := policyContext.NewResource()
+	oldResource := policyContext.OldResource()
+	emptyResource := unstructured.Unstructured{}
+
+	if rule.MatchResources.All != nil || rule.MatchResources.Any != nil {
+		matched := match.CheckMatchesResources(
+			policyContext.OldResource(),
+			kyvernov2beta1.MatchResources{
+				Any: rule.MatchResources.Any,
+				All: rule.MatchResources.All,
+			},
+			make(map[string]string),
+			kyvernov2.RequestInfo{},
+			oldResource.GroupVersionKind(),
+			"",
+		)
+		if matched != nil {
+			return nil, nil
+		}
+	}
+
+	if rule.ExcludeResources.All != nil || rule.ExcludeResources.Any != nil {
+		excluded := match.CheckMatchesResources(
+			policyContext.OldResource(),
+			kyvernov2beta1.MatchResources{
+				Any: rule.ExcludeResources.Any,
+				All: rule.ExcludeResources.All,
+			},
+			make(map[string]string),
+			kyvernov2.RequestInfo{},
+			oldResource.GroupVersionKind(),
+			"",
 		)
+		if excluded == nil {
+			return nil, nil
+		}
+	}
+	if err := policyContext.SetResources(emptyResource, oldResource); err != nil {
```

Comment:
I imagine this could be an issue if we enable concurrent policy application 🤔

Reply:
yes, but we don't have a design for concurrent policy execution at the moment, I will update that logic in that implementation 

We should definitely keep this in mind while designing concurrency safe policy context